### PR TITLE
[WIP] Custom Status Mobile UI

### PR DESCRIPTION
**Pull request by wreiske**
@RocketChat/ReactNative

This is a work in progress. Any help is greatly appreciated!

Here are some sample images so far:

636Amfvuo8

Screenshot_1567669035

explorer_S985cdk1ke
Screenshot_1567668896
Screenshot_1567667858

Screenshot_1567670521
Screenshot_1567670517
Screenshot_1567670627

Screenshot_1567671641

### CODE REVIEW

**Comment by wreiske** [comment]
Things that need to be figured out:

- [ ] It doesn't look like the 'user-status' events are updating / or there's something needed to redraw the statuses. They aren't changing if you change the status on another client.
- [ ] Figure out why the status isn't pulling in on the profile page to edit
- [ ] The status is currently showing as the current user in every channel. Need to figure out how to show the room user's status instead of the current logged in user.

File: app/views/RoomView/index.js [hunk 1]
```diff
@@ -624,6 +625,11 @@ class RoomView extends React.Component {
 		return (
 			<SafeAreaView style={styles.container} testID='room-view' forceInset={{ vertical: 'never' }}>
 				<StatusBar />
+				{this.t === 'd' && user && user.statusText ? (
```
Comment (djorkaeffalexandre) [inline app/views/RoomView/index.js]: create a function like a `renderStatusText()` :)

**Comment by wreiske** [comment]
I have a strict deadline that I must hit this week, so it will be later in the week before I will have time to mess with this again. If anyone wants to take a stab at it, go ahead. It'll be Thursday or Friday before I go at it again. Thanks!

**Comment by Prateek93a** [comment]
@wreiske @diegolmello @djorkaeffalexandre Hello, I would like to show my work on this feature. I used all the code that @wreiske  had written to implement the above mentioned feature. Most of the work was already done. I added my code to solve the 3 issues mentioned above, added change status modal and added theme support and resolved the conflicts. Also I thought it would be better not to show status text at the top of the room view, so I removed it but it can be added again if needed. The gif is shown below. Does it look fine?

ezgif com-crop(14)

**Comment by wreiske** [comment]
Thanks for updating and fixing the issues described above. Can you open a pull request to merge into wreiske:custom-status-ui? I will look at the changes and merge them into this PR. 

I would rather (IMO) see the status on the room view if it's a direct message, there's room there to have it, so why not show it? People aren't going to click into user's profiles to see their status -- but they will see it if it's in the room header.

**Comment by Prateek93a** [comment]
@wreiske Yes you are right. Now I get it, status should be visible on the room view as well so that people can see it otherwise not everyone will look into the room info to see status. Thanks.
Also,I am  a little confused here, how should I raise Pr to merge into wreiske:custom-status-ui? Should I raise Pr in your forked repo?

**Comment by wreiske** [comment]
That would work, yes. I will review it in the next 2 business days and get it merged in. That will update this PR.

**Comment by Prateek93a** [comment]
I tried to do that. First I tried to fork your repo but that github gave message that I already had a forked repo of Rocket.Chat.ReactNative.Then I tried to directly raise Pr to your forked repo, but that gave access denied error.

**Comment by wreiske** [comment]
I created a PR. It _should_ automatically update as you push changes to your branch.

https://github.com/wreiske/Rocket.Chat.ReactNative/pull/1

**Comment by Prateek93a** [comment]
Ok Thanks

**Comment by wreiske** [comment]
@Prateek93a if you can take a look at why the status modal isn't refreshing the latest status, that'd be great. thanks!

- Change status on mobile app.
- Change status on desktop
- Tap to change status on mobile app to see the original status message in the modal.

Screen Shot 2020-01-25 at 2 55 41 PM

**Comment by Prateek93a** [comment]
@wreiske Yes, surely.  I will try to fix by tomorrow .

**Comment by wreiske** [comment]
> @wreiske Yes, surely. I will try to fix by tomorrow .

Also, the original bugs still exist... The status message does not refresh in the mobile app when changed on desktop. You have to click out of the room and click back in to see the change.

image

It also looks like this introduced some bugs. When there is no status message, the status bar shouldn't be displayed in a DM.
image

The status also doesn't refresh in the user profile:
image

image

Thanks for all the help!

**Comment by Prateek93a** [comment]
Ok I will look into the bug, and also make the changes that you suggested.

**Comment by Prateek93a** [comment]
> @Prateek93a if you can take a look at why the status modal isn't refreshing the latest status, that'd be great. thanks!
> 
>     * Change status on mobile app.
> 
>     * Change status on desktop
> 
>     * Tap to change status on mobile app to see the original status message in the modal.
> 
> 
> Screen Shot 2020-01-25 at 2 55 41 PM

@wreiske  I have fixed the above bug, and also added the code to hide the status text bar when there is no status. Now I am working on the last bug where the room status was not getting updated. Also I tried pushing my commits, but for that I had to pull the changes first. But when I pulled, the code was still having duplicates, and fixed color. I am not sure why this happened. So shall I remove them and then push the commits?

**Comment by wreiske** [comment]
It may be best to fork my repo, edit the custom status ui branch, and then make a pull request back. If you fork my repo on the branch this PR is, it shouldn't have any duplicates. That way you run another PR to my fork, I'll review and test, and then merge it and this PR will update.

**Comment by Prateek93a** [comment]
Actually I tried to do that but since I already have a forked this repo, so I am not able to fork your repo. 
I have received the latest commit you made i.e. `merge pull request` but in that, there was no commit to remove the duplicate code. I think if you push the `remove duplicates` commit again, then after pulling the changes I shall receive those changes.

**Pull request closed without merging**